packages/synapse-react-client/src/components/GenericCard/CroissantButton/CroissantButton.tsx

```diff
@@ -0,0 +1,47 @@
+import GenericCardActionButton from '@/components/GenericCard/GenericCardActionButton'
+import ConditionalWrapper from '@/components/utils/ConditionalWrapper'
+import { useGetCroissantUrl } from '@/synapse-queries/croissant/useGetCroissantUrl'
+import Skeleton from '@mui/material/Skeleton'
+
+type CroissantButtonProps = {
+  /** The synID of the dataset */
+  datasetId: number
+  /** The version number of the dataset */
+  datasetVersionNumber: number
+}
+
+/**
+ * Given a Synapse ID and version number of a dataset, this component will render a button that links to the Croissant
+ * metadata for that dataset.
+ */
+function CroissantButton(props: CroissantButtonProps) {
+  const { datasetId, datasetVersionNumber } = props
+
+  const { data: croissantUrl, isLoading } = useGetCroissantUrl(
+    datasetId,
+    datasetVersionNumber,
+  )
+
+  if (!croissantUrl && !isLoading) {
+    return null
+  }
+
+  return (
+    <ConditionalWrapper
+      condition={isLoading}
+      wrapper={Skeleton}
```

Comment:
Does this mean that every card will have a loading skeleton button, and then nothing (if a croissant row is not found)?

Reply:
The loading spinner will only be seen while the Croissant mapping table is initially fetched; the client-side filter function is synchronous.